CONTRIBUTING.md

```diff
@@ -0,0 +1,70 @@
+# Contributing to Cloud Native Patterns
+We love your input! We want to make contributing to this project as easy and transparent as possible, whether it's:
+
+- Reporting a bug
+- Discussing the current state of the patterns
+- Submitting a fix
+- Proposing new patterns
+- Becoming a maintainer
+
+## We Develop with Github
+We use github to host code, to track issues and feature requests, as well as accept pull requests.
+
+## We Use [Github Flow](https://guides.github.com/introduction/flow/index.html), So All Code Changes Happen Through Pull Requests
+Pull requests are the best way to propose changes to the patterns. We actively welcome your pull requests:
+
+1. Fork the repo and create your branch from `master`.
+2. If you've added pattern, double check for typos
+3. If you've changed pattern, explain change in the PR comment
+5. Make sure your patterns are following the structure of the pattern
+6. Issue that pull request!
+
+## Any contributions you make will be under the `Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License`
+In short, when you submit pattern changes, your submissions are understood to be under the same [CC 4.0 License](https://creativecommons.org/licenses/by-nc-sa/4.0/) that covers the project. Feel free to contact the maintainers if that's a concern.
+
+## Report bugs using Github's issues
+We use GitHub issues to track public bugs. Report a bug by [opening a new issue](https://github.com/ContainerSolutions/cloud-native-patterns/issues/new/choose).
+
+## Write issue reports with detail, background and explanation
+
+**Great Issue Reports** tend to have:
+
+- A quick summary and/or background
+- What you expected should be included in the pattern
+- Notes (possibly including why you think this might be needed)
+
+## Use a Consistent Pattern Layout
+The pattern should use the following layout:
+```markdown
+
+---
+title: <name of the pattern>
+description: <description>
+layout: pattern
+---
+
+![<name of the pattern>]({{ site.baseurl }}/assets/images/generic.png) //link to image if relevant
+
+<situation description>
+
+## In This Context
+
+<set context>
```

Comment:
This section actually describes the problems and pitfalls which are found within this context. So maybe <problems within context>?